prov/sockets/src/sock_conn.c

```diff
@@ -422,7 +422,7 @@ struct sock_conn *sock_ep_connect(struct sock_ep_attr *ep_attr, fi_addr_t index)
 
 	ret = connect(conn_fd, (struct sockaddr *) addr, sizeof *addr);
 	if (ret < 0) {
-		if (errno == EINPROGRESS) {
+		if (ofi_is_progress_socket()) {
```

Comment:
Okay - I think I see how this is being used.  What if we defined 'sockerr' that can map to errno on unix, and something like this on windows:

```
static inline int sockerr(void)
{
    int wsaerror = WSAGetLastError();
    switch (wsaerror) {
    case: WSAEINPROGRESS:
    case WSAEWOULDBLOCK:
        return EINPROGRESS;
    default:
        return wsaerror;
}